src/RA_Dlg_MemBookmark.h

```diff
@@ -3,42 +3,92 @@
-    void SetDescription(const std::wstring& string) { m_sDescription = string; }
-    void SetAddress(unsigned int nVal) noexcept { m_nAddress = nVal; }
-    void SetType(unsigned int nVal) noexcept { m_nType = nVal; }
-    void SetValue(unsigned int nVal) noexcept { m_sValue = nVal; }
-    void SetPrevious(unsigned int nVal) noexcept { m_sPrevious = nVal; }
-    void IncreaseCount() noexcept { m_nCount++; }
-    void ResetCount() noexcept { m_nCount = 0; }
-
-    void SetFrozen(bool b) noexcept { m_bFrozen = b; }
-    void SetDecimal(bool b) noexcept { m_bDecimal = b; }
-
-    inline const std::wstring& Description() const noexcept { return m_sDescription; }
-    unsigned int Address() const noexcept { return m_nAddress; }
-    unsigned int Type() const noexcept { return m_nType; }
-    unsigned int Value() const noexcept { return m_sValue; }
-    unsigned int Previous() const noexcept { return m_sPrevious; }
-    unsigned int Count() const noexcept { return m_nCount; }
-
-    bool Frozen() const noexcept { return m_bFrozen; }
-    bool Decimal() const noexcept { return m_bDecimal; }
+public: // Too much trouble making this a literal type
+    MemBookmark() noexcept = default;
+    ~MemBookmark() noexcept = default;
+    // Copying deleted to enforce uniqueness and speed
+    MemBookmark(const MemBookmark&) = delete;
+    MemBookmark& operator=(const MemBookmark&) = delete;
+    inline constexpr MemBookmark(MemBookmark&&) noexcept = default;
+    inline constexpr MemBookmark& operator=(MemBookmark&&) noexcept = default;
+
+    inline auto SetDescription(const std::wstring& string) { m_sDescription = string; }
+    // rvalues are sometimes used
+    inline auto SetDescription(std::wstring&& string) noexcept { m_sDescription = std::move(string); }
+    _CONSTANT_FN SetAddress(unsigned int nVal) noexcept { m_nAddress = nVal; }
+    _CONSTANT_FN SetType(unsigned int nVal) noexcept { m_nType = nVal; }
+    _CONSTANT_FN SetValue(unsigned int nVal) noexcept { m_sValue = nVal; }
+    _CONSTANT_FN SetPrevious(unsigned int nVal) noexcept { m_sPrevious = nVal; }
+    _CONSTANT_FN IncreaseCount() noexcept { m_nCount++; }
+    _CONSTANT_FN ResetCount() noexcept { m_nCount = 0; }
+
+    _CONSTANT_FN SetFrozen(bool b) noexcept { m_bFrozen = b; }
+    _CONSTANT_FN SetDecimal(bool b) noexcept { m_bDecimal = b; }
+
+    _NODISCARD inline auto& Description() const noexcept { return m_sDescription; }
+    _NODISCARD _CONSTANT_FN Address() const noexcept { return m_nAddress; }
+    _NODISCARD _CONSTANT_FN Type() const noexcept { return m_nType; }
+    _NODISCARD _CONSTANT_FN Value() const noexcept { return m_sValue; }
+    _NODISCARD _CONSTANT_FN Previous() const noexcept { return m_sPrevious; }
+    _NODISCARD _CONSTANT_FN Count() const noexcept { return m_nCount; }
+
+    _NODISCARD _CONSTANT_FN Frozen() const noexcept { return m_bFrozen; }
+    _NODISCARD _CONSTANT_FN Decimal() const noexcept { return m_bDecimal; }
 
 private:
     std::wstring m_sDescription;
```

Comment:
We could make this a literal type of we made this `const wchar_t*`, but it seems to cause problems.